### PR TITLE
feat#103: replace console.logs with Nest Logger

### PR DESCRIPTION
**Pull request by Thistle-Strum**
🔥 Summary
Adds NestJs Logger 
Screenshot 2023-06-23 at 11 55 30 AM
Screenshot 2023-06-23 at 11 54 59 AM
Screenshot 2023-06-23 at 12 00 10 PM

### CODE REVIEW

File: app/src/blocks/blocks.service.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { ApiError } from "src/utils/ApiError";
 
 @Injectable()
 export class BlocksService {
+  readonly logger = new Logger(BlocksService.name);
```
Comment (mgibson-casper) [inline app/src/blocks/blocks.service.ts:13]: Is there not a way to inject this into the service via the constructor?

File: app/src/gateway/gateway.service.ts [hunk 2]
```diff
@@ -20,21 +21,21 @@ export class GatewayService
   io: Namespace;
 
   afterInit(): void {
-    console.log("Websocket gateway initialized.");
+    Logger.log("Websocket gateway initialized.");
```
Comment (mgibson-casper) [inline app/src/gateway/gateway.service.ts]: Why are we accessing `.log` on `Logger` directly here as opposed to defining an instance of the class like in blocks service before accessing it?

File: app/src/blocks/blocks.service.ts (hunk 1 above)
Comment (DorianKwan) [reply app/src/blocks/blocks.service.ts:13]: I took a little bit of time to read into the docs.  What David is doing here seems to be the most simple approach suggested by their docs.  It's quite impressive how non-opinionated NestJS is.

Looks like we can use the Logger class as David is doing here wherever we'd like -- but there is not defined/consistent pattern.  Seems like a logger comes shipped out of the box using a singleton pattern or we can inject a new instance in each module/service.  Dep injection seems to require us to create a custom logger class with the `@injectable` decorator.

I think we should future proof with a custom logger that's injected.  Thoughts @mgibson-casper ?

File: app/src/gateway/gateway.service.ts (hunk 2 above)
Comment (DorianKwan) [reply app/src/gateway/gateway.service.ts]: +1, whatever pattern/approach we take we should be using the instanced logger

**Review by DorianKwan — CHANGES_REQUESTED** [review]

File: app/src/blocks/blocks.service.ts (hunk 1 above)
Comment (mgibson-casper) [reply app/src/blocks/blocks.service.ts:13]: Well we are applying the `Injectable` here and in all other service classes using the logger. I also just took a quick look and this seems OK to me so I'm good to keep as is 👍

File: app/src/gateway/gateway.service.ts (hunk 2 above)
Comment (Thistle-Strum) [reply app/src/gateway/gateway.service.ts]: Updated!  
Screenshot 2023-06-26 at 12 12 16 PM

**Comment by mgibson-casper** [comment]
Hey @Thistle-Strum have you tested this to make sure it logs properly? The top of this page https://docs.nestjs.com/techniques/logger has a code snippet for adding this to the app creation method:

```
const app = await NestFactory.create(AppModule, {
  logger: ['error', 'warn'],
});
```

Do we not need this?

**Comment by DorianKwan** [comment]
> Hey @Thistle-Strum have you tested this to make sure it logs properly? The top of this page https://docs.nestjs.com/techniques/logger has a code snippet for adding this to the app creation method:
> 
> ```
> const app = await NestFactory.create(AppModule, {
>   logger: ['error', 'warn'],
> });
> ```
> 
> Do we not need this?

I don't believe he does -- but it might be a good idea to run in debug more on dev environments

**Review by DorianKwan — APPROVED** [review]

**Review by mgibson-casper — APPROVED** [review]

**Comment by Thistle-Strum** [comment]
I read that portion of the docs as the most basic _customization_ allowing you to ask the logger to only provide specified log levels (i.e. `'log' | 'error' | 'warn' | 'debug' | 'verbose'`), so I didn’t add it to main.  I approached each instance as unique and followed the example @ [Using the logger for application logging](https://docs.nestjs.com/techniques/logger#using-the-logger-for-application-logging).

**Pull request merged**